great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -664,9 +665,9 @@ def get_domain_records(  # noqa: C901, PLR0912, PLR0915
         row_condition = domain_kwargs.get("row_condition", None)
         if row_condition:
             condition_parser = domain_kwargs.get("condition_parser", None)
-            if condition_parser == "spark":
+            if condition_parser == ConditionParser.SPARK:
                 data = data.filter(row_condition)
-            elif condition_parser == "great_expectations__experimental__":
+            elif condition_parser == ConditionParser.GX:
```

Comment:
My idea was to also allow `ConditionParser.GX_DEPRECATED` here, so users can continue to pass the old string without breaking.

Reply:
Got it, that makes sense 😬

Reply:
Do we have a deprecation strategy? How long do we keep these types of paths around?

Reply:
We follow semver, so it wouldn't be until GX 2.0 in theory.